lib/webdriver.py

```diff
@@ -176,35 +228,37 @@ def _login_listener(self, data: JSON) -> None:
             logger.debug("Upcoming trips response has been received")
             self.trips_request_id = data["params"]["requestId"]
 
-    def _wait_for_attribute(self, attribute: str) -> None:
-        logger.debug("Waiting for %s to be set (timeout: %d seconds)", attribute, WAIT_TIMEOUT_SECS)
+    def _wait_for_attribute(
+        self, attribute: str, timeout: Optional[int] = WAIT_TIMEOUT_SECS
+    ) -> None:
+        logger.debug("Waiting for %s to be set (timeout: %d seconds)", attribute, timeout)
         poll_interval = 0.5
 
-        attempts = 0
-        max_attempts = WAIT_TIMEOUT_SECS / poll_interval
-        while not getattr(self, attribute) and attempts < max_attempts:
```

Comment:
Personally, I find the `while` loop more readable than the `for` loop due to the number of indentations.